sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -261,25 +258,199 @@ def _is_json_content_type(content_type: str) -> bool:
+                    )
+                    if updated_sentinel is not None:
+                        logging.debug(
+                            "Refresh all triggered by key: %s label %s.",
+                            key,
+                            label,
+                        )
+                        self._load_all(**kwargs)
+                        # TODO: We need to reset all etags, not just the one that triggered it, could cause extra refreshes.
+                        self._refresh_on[(key, label)] = updated_sentinel.etag
+                        self._refresh_timer.reset()
+                        return
+        except (ServiceRequestError, ServiceResponseError) as e:
+            logging.debug("Failed to refresh, retrying: %r", e)
+            self._refresh_timer.retry()
+        except HttpResponseError as e:
+            # There might be specific status codes that we want to silently backoff and retry.
+            # Need more service specific details here, raising for now.
+            if _is_retryable_error(e):
+                self._refresh_timer.retry()
+            if self._on_refresh_error:
+                self._on_refresh_error(e)
+                return
+            raise
+        except Exception as e:
+            if self._on_refresh_error:
+                self._on_refresh_error(e)
+                return
+            raise
+
+    def _load_all(self, **kwargs):
+        configuration_settings = {}
+        for select in self._selects:
+            configurations = self._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            for config in configurations:
+                key = self._proccess_key_name(config)
+                value = self._proccess_key_value(config)
+
+                if isinstance(config, FeatureFlagConfigurationSetting):
+                    feature_management = self._dict.get(FEATURE_MANAGEMENT_KEY, {})
+                    feature_management[key] = value
+                    if FEATURE_MANAGEMENT_KEY not in self.keys():
+                        configuration_settings[FEATURE_MANAGEMENT_KEY] = feature_management
+                else:
+                    configuration_settings[key] = value
+                # Every time we run load_all, we should update the etag of our refresh sentinels
+                # so they stay up-to-date.
+                # Sentinel keys will have unprocessed key names, so we need to use the original key.
+                if (config.key, config.label) in self._refresh_on:
+                    self._refresh_on[(config.key, config.label)] = config.etag
+
+        # Let's check whether any of the settings configured for refresh don't actually exist
+        # This should only happen during the initial `load_all`, not subsequent refreshes.
+        no_etag = [(key, label) for (key, label), etag in self._refresh_on.items() if etag is None]
```

Comment:
Looks like lines 414-421 can be removed - I think this was only used by the old "refresh_only" feature.